swri_roscpp/include/swri_roscpp/node_handle.h

```diff
@@ -202,6 +202,34 @@ class NodeHandle
     return set;
   }
 
+  inline bool getParam(const std::string& name, std::vector<std::string>& value,
+    const std::string description)
+  {
+    std::string real_name = resolveName(name);
+    bool set = nh_->pnh_.getParam(real_name, value);
+    ROS_INFO("Read parameter %s", name.c_str());
+
+    if (shouldAddParameter(real_name))
```

Comment:
If the parameter isn't set, should we even run all of this code?

Reply:
That is a valid point, the XmlRpcValue getParam function has the same structure as this vector<std::string> getParam, so I am unsure what we want to do here.

Reply:
Nevermind ^, we do want to run all of this code. This swri implementation of node_handle documents the parameters we are using in our code. We would especially want to document params if they aren't set, so that when the rosman check script (the issue you merged earlier) runs, it can see that there is a parameter that is documented but not set, and output that.